### ISSUE TITLE
Drop-Source not observed if Host-Computer is offline at startup of JMM Server

### ISSUE THREAD

**Issue by yhoogi** [japanesemediamanager/jmmserver#495]
Setting
Drop-Source: Network Folder on a different host computer;  Watch for new files enabled
At startup of JMM server Drop-Source is not available (Host Computer offline)

Effect:
Drop-Source is not monitored, even if Host Computer comes online later
New files in Drop-Source are not recognised and processed

To resolve: Have to restart JMM server

**Comment by maxpiva** [comment]
This is a Know issue (at least by me :P) if you disconnect and connect a NAS/USB Drive, etc the behavior is the same. The FileSystemWatcher provided with the .NET Framework after initialized, if you disconnect the drive in question and connect it again. The FileSystem Events will not longer be triggered. There is a stackoverflow question about this:

[FileSystemWatcher Network Disconnect](http://stackoverflow.com/questions/9161993/filesystemwatcher-network-disconnect)

We have to make the FileSystemWatcher resilent to network and disconnect/connect events.

**Comment by maxpiva** [comment]
For Our Info, FileSystemWatcher replacement.

[Tamed FileSystemwatcher](https://petermeinl.wordpress.com/2015/05/18/tamed-filesystemwatcher/)

**Comment by hidden4003** [comment]
Tested with win 10 laptop and win 10 vm, doesn't work.
Test used was:
1. Make sure vm offline, server started, message in log about folder being watched.
2. Bring VM online, make sure folder is accesible, drop some file.
Result - nothing happens.
Restart server, drop file, file gets hashed imported.

**Comment by yhoogi** [comment]
Will this be corrected with v3.8?

Cheers,

**Comment by da3dsoul** [comment]
Nope! As far as I am aware, it is an issue with the system we use. @maxpiva you posted a possible solution further up. Is that a feasible solution?

**Comment by maxpiva** [comment]
We already implemented that, i think somehow server is missing a call in StartWatchingFiles.

We need to call for every fsw, the Start Method
That will start a timer, that will check the state mount point periodically, and start when needed, i cannot commit any changes right now. (Not in the work computer).
